lib/dart.yaml

```diff
@@ -1,194 +1,209 @@
-    - member-ordering:
-        order:
-          - public-fields
-          - private-fields
-          - public-getters
-          - private-getters
-          - public-setters
-          - private-setters
-          - constructors
-          - close-method
-          - dispose-method
-          - public-methods
-          - private-methods
-        widgets-order:
-          - constructor
-          - named-constructor
-          - const-fields
-          - final-fields
-          - init-state-method
-          - did-change-dependencies-method
-          - did-update-widget-method
-          - dispose-method
-          - private-methods
-          - overriden-public-methods
-          - build-method
-    - move-records-to-typedefs:
-        min-fields: 3
-        min-occurences: 2
-    - move-variable-closer-to-its-usage
-    - move-variable-outside-iteration
-    - newline-before-return
-    - no-boolean-literal-compare
-    - no-empty-block
-    - no-equal-arguments:
-        ignored-parameters:
-          - height
-          - width
-    - no-equal-conditions
-    - no-equal-nested-conditions
-    - no-equal-switch-case
-    - no-equal-switch-expression-cases
-    - no-equal-then-else
-    - no-magic-number:
-        allowed: [ -1, 0, 1 ]
-    - no-object-declaration
-    - prefer-addition-subtraction-assignments
-    - prefer-any-or-every
-    - prefer-correct-callback-field-name:
-        name-pattern: ^on[A-Z]+
-    - prefer-correct-error-name:
-        allowed-names: [ error, exception, failure ]
-    - prefer-correct-identifier-length:
-        max-identifier-length: 30
-        min-identifier-length: 2
-    - prefer-correct-stream-return-type
-    - prefer-declaring-const-constructor:
-        ignore-abstract: true
-    - prefer-early-return:
-        ignore-if-case: true
-    - prefer-explicit-function-type
-    - prefer-explicit-type-arguments
-    - prefer-immediate-return
-    - prefer-match-file-name:
-        exclude:
-          - "test/**"
-          - "lib/**/*wm.dart"
-    - prefer-moving-to-variable:
-        allowed-duplicated-chains: 3
-    - prefer-named-boolean-parameters:
-        ignore-single: true
-        ignore-single-boolean: false
-    - prefer-null-aware-spread
-    - prefer-parentheses-with-if-null
-    - prefer-public-exception-classes
-    - prefer-return-await
-    - prefer-simpler-patterns-null-check
-    - prefer-switch-with-enums
-    - prefer-test-matchers
-    - prefer-typedefs-for-callbacks
-    - prefer-unique-test-names
-    - prefer-wildcard-pattern
-    # Style
-    - unnecessary-trailing-comma: false
+    rules:
+        # https://dcm.dev/docs/rules/common
+        - arguments-ordering:
+              last:
+                  - body
+                  - child
+                  - children
+        - parameters-ordering:
+              required: "first"
+              default: "last"
+              super: "last"
+
+        - avoid-accessing-collections-by-constant-index
```

Comment:
How we'll be process such cases?
```dart
  ...
  final someString = 'some-decoded-id-in-the-middle';
  final parts = someString.split('-');
  assert(parts.length == 6, 'Incorrect format');
  return parts[2];
}

Reply:
Oops, it seems the Yaml file was formatted, so everything got into the diff... 
But I didn’t change the rules themselves in this pr.

Reply:
Maybe create an issue?